src/main/java/org/testng/xml/DefaultXmlWeaver.java

```diff
@@ -0,0 +1,275 @@
+package org.testng.xml;
+
+import org.testng.reporters.XMLStringBuffer;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.testng.collections.CollectionUtils.hasElements;
+import static org.testng.internal.Utils.isStringNotEmpty;
+import static org.testng.xml.XmlSuite.*;
+
+/**
+ * This class provides String representation of both {@link XmlSuite} and {@link XmlTest} but adds
+ * an XML comment as the test name and  suite name at the end of the corresponding tags.
+ */
+class DefaultXmlWeaver implements IWeaveXml {
+
+    private String defaultComment;
```

Comment:
`final`

Reply:
This class cannot be a `final` class because `org.testng.xml.CommentDisabledXmlWeaver` extends it.

Reply:
I was talking about `defaultComment`, not the class ;)

Reply:
Ah ok. I didn't realize that.